### PR TITLE
Struggling a lot

### PR DESCRIPTION
**Pull request by PikaJoyce**
I'm having a bunch of issues trying to get the drop down to work as it should but Steam's interface is giving me a pretty hard time with trying to position it properly to trying to work with the overflow. This is what I got so far and the hovering works but the drop downs are.... let us say interesting. I'm pretty blocked on this so any other insights to get this working is welcome.

What I've done today :
- Got drop down containers in and on hover changes color.
- We finally figured out where the steam smol arrow lives.
- Can conditionally render SOMETHING but not how I want it to.
- Lots of experimenting with many many failures.

What I need to do:
- Get the drop down tables to display properly with effects and all
- Be able to select an option from the table and then change the item selected in the container

Higher level goals:
- Be able to send all this information back to the parent component
- Write tests that reflect that my component is functional

What I need help with:
EVERYTHING.

Here's a screen shot of what it's currently doing. There are options beneath the "No" but we can't see it. I've adjusted the overflows and tried fiddling around with it and I've only found it in several components and when those were either removed or adjusted (to a overflow:Scroll or w/e) the entire component just twisted in weird places it shouldn't.
 
Screen Shot 2019-05-24 at 8 50 20 PM

### CODE REVIEW

**Comment by PikaJoyce** [comment]
Still having issues with axios and the text editor not liking it. However, I connected the server to the client successfully despite VSCode telling me I'm wrong. The drop down tables exist and I spent a few hours trying to get it to append on top of the entire component but wasn't able to successfully do that. Otherwise both drop downs "work" 

Screen Shot 2019-05-25 at 10 01 22 PM

I've adjusted the schema since I only took in the inputs that are being entered into my own component. The username/avatar and what have you are out of scope from my component so maybe I can add that tomorrow? 

I still need to get tests up and running since the functionality is all there. 
Here's a short demo :
[DEMO](https://gfycat.com/ElasticColorlessBasil)

### TL;DR
**What I did and what works?**
- _Most_ things are functional
- The exception would probably be the drop down table which I believe needs to be appended to the entire app's body rather than on my own component.
- Can send post request to the backend
- Adjusted the schema

**What to work on?**
- LOL, fixing the gosh darn drop down menu, which I've come to learn isn't actually a drop down at all but a menu that straight up appears on top of the page.
- Writing unit tests for both the front and back end 

**What I need help with**
- For now, nothing. 
- Potentially the drop down table and tests but that's about it.

File: client/src/components/Controls.jsx [hunk 1]
```diff
@@ -40,30 +43,167 @@ const AllowComments = styled.div`
...
     return (
       <ReviewControls>
         <ControlBlock>
-          <StyledSpan className="controlText"> Visibility :</StyledSpan>
-          <label> DROP DOWN BOX HERE LOL</label>
+          <StyledSpan className="controlText"> Visibility :</StyledSpan> &nbsp;
+          <DropDownContainer onClick={e => this.handleVisibilityDrop(e)}>
+            <DropDownA>{this.state.visibility}</DropDownA>
+            {this.state.showMenuPublic ? (
+              <VisibilityDropDown
```
Comment (eric-do) [inline client/src/components/Controls.jsx:180]: Your VisibilityDropDown component and LanguageDropDown components are basically the same, so it's likely when you update one you'll need to update the other. Try to refactor so from your Controls component you can call them similar this:

```javascript
<MenuDropDown
                options={this.state.visibilityOptions}
                updateSelected={this.updateVisibility}
/>
<MenuDropDown
                options={this.state.languages}
                updateSelected={this.updateLanguages}
 />
```

Then in the merged component maybe do something like this:
```javascript
  constructor(props) {
    super(props);
    this.state = {
      selected: ''
    };
  }
```

**Review by eric-do — APPROVED** [review]
It's coming along! 

I don't get the same errors you did when I ran your code, but I did get some lint errors on database/index.js - just missing a space after a couple commas - nothing critical. I did see new records get added to db.

Now that you're wrapping up, a couple things to go look back on :)
1. Refactor dropdown (I left inline comment in code)
2. Resolve the open/close menu with focus/blur

I actually continued to look into 2 a little today as well and realized it has some real world value for accessibility. If people are filling out forms, they need to be able to tab through them - a focus/blur method would be important for that.

Found a couple suggested implementations on Stack Overflow for it:
- https://stackoverflow.com/a/44378829/1397557
- https://stackoverflow.com/a/45323523/1397557

File: client/src/components/Controls.jsx (hunk 1 above)
Comment (PikaJoyce) [reply client/src/components/Controls.jsx:180]: I'll look more into that in the AM. Thanks for the suggestions Dai Lo! I wasn't entirely sure what the onBlur thing did so thanks for explaining that.
